python/tests/splines_utils.py

```diff
@@ -0,0 +1,851 @@
+        raise RuntimeError('PEtab lint failed')
+
+    # Write PEtab problem to disk
+    if folder is None:
+        return problem, initial_values, T
+    folder = os.path.abspath(folder)
+    os.makedirs(folder, exist_ok=True)
+    problem.to_files(
+        sbml_file=os.path.join(folder, f'{model_name}_model.xml'),
+        condition_file=os.path.join(folder, f'{model_name}_conditions.tsv'),
+        measurement_file=os.path.join(folder,
+                                      f'{model_name}_measurements.tsv'),
+        parameter_file=os.path.join(folder, f'{model_name}_parameters.tsv'),
+        observable_file=os.path.join(folder,
+                                     f'{model_name}_observables.tsv'),
+        yaml_file=os.path.join(folder, f'{model_name}.yaml'),
+    )
+    return os.path.join(folder, f'{model_name}.yaml'), initial_values, T
+
+
+def simulate_splines(
+        splines,
+        params_true,
+        initial_values=None,
+        *,
+        folder: Optional[str] = None,
+        keep_temporary: bool = False,
+        benchmark: Union[bool, int] = False,
+        rtol: float = 1e-12,
+        atol: float = 1e-12,
+        maxsteps: int = 500_000,
+        discard_annotations: bool = False,
+        use_adjoint: bool = False,
+        skip_sensitivity: bool = False,
+        petab_problem=None,
+        amici_model=None,
+        **kwargs
+):
+    """
+    Create a PEtab problem using `create_petab_problem` and simulate it with
+    AMICI.
+
+    :param splines:
+        passed to `create_petab_problem`
+
+    :param params_true:
+        passed to `create_petab_problem`
+
+    :param initial_values:
+        passed to `create_petab_problem`
+
+    :param folder:
+        working directory (a temporary one is used if not specified)
+
+    :param keep_temporary:
+        whether to keep or delete temporary working directories on exit
+
+    :param benchmark:
+        instead of returning the simulation data, run the simulation
+        `benchmark` times (defaults to `50` if `benchmark` is `True`)
+        and return execution times
+
+    :param rtol:
+        relative tolerance for AMICI solver
+
+    :param atol:
+        absolute tolerance for AMICI solver
+
+    :param maxsteps:
+        maximum number of steps for AMICI solver
+
+    :param discard_annotations:
+        whether to discard spline annotations,
+        forcing AMICI to read the spline as a piecewise assignment rule
+
+    :param use_adjoint:
+        whether to use adjoint sensitivity computation
+
+    :param skip_sensitivity:
+        whether to skip sensitivity computation
+
+    :param petab_problem:
+        PEtab problem (if already created)
+
+    :param amici_model:
+        AMICI model (if already created)
+
+    :param kwargs:
+        passed to `create_petab_problem`
+    """
+    # If no working directory is given, create a temporary one
+    if folder is None:
+        if keep_temporary:
+            folder = TemporaryDirectory().name
```

Comment:
Yes, it is not guaranteed to work. But it's only if you have to debug the thing and look at the generated code. And in practice it works. It could be definitely be made to be guaranteed to work, but it would be overcomplicated for what it is actually used for. Unless you have in mind a concise alternative.